File: src/main/java/net/jeikobu/jbase/config/AbstractConfigManager.java
package net.jeikobu.jbase.config;

import sx.blah.discord.handle.obj.IGuild;

import java.util.Locale;

public abstract class AbstractConfigManager {
    public abstract IGlobalConfig getGlobalConfig();
    public abstract AbstractGuildConfig getGuildConfig(IGuild guild);

    public Locale getLocale(IGuild guild) {
        return getGuildConfig(guild).getGuildLocale().orElse(getGlobalConfig().getGlobalLocale());
    }

    public String getCommandPrefix(IGuild guild) {
        return getGuildConfig(guild).getCommandPrefix().orElse(getGlobalConfig().getDefaultCommandPrefix());
    }
}

File: src/main/java/net/jeikobu/jbase/config/AbstractConfigManager.kt
package net.jeikobu.jbase.config

import net.dv8tion.jda.core.entities.Guild
import net.jeikobu.jbase.impl.config.VolatileStorage
import java.util.*

abstract class AbstractConfigManager {
    val volatileStorage = VolatileStorage()

    abstract val globalConfig: IGlobalConfig

    fun getGuildConfig(guild: Guild): AbstractGuildConfig {
        return getGuildConfig(guild, globalConfig)
    }

    protected abstract fun getGuildConfig(guild: Guild, globalConfig: IGlobalConfig): AbstractGuildConfig

    fun getLocale(guild: Guild): Locale {
        return getGuildConfig(guild).guildLocale ?: globalConfig.globalLocale
    }

    fun getCommandPrefix(guild: Guild): String {
        return getGuildConfig(guild).commandPrefix ?: globalConfig.defaultCommandPrefix
    }
}